### PR TITLE
Hcap 932 indigenous filter

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Added the indigenous filter for the employer portal 
Added row level filtering
https://freshworks.atlassian.net/browse/HCAP-932
Screen Shot 2021-11-08 at 1 41 57 PM

### CODE REVIEW

File: server/package-lock.json [hunk 1]
```diff
@@ -1,10932 +1,8 @@
 {
   "name": "health-career-access-program-server",
   "version": "1.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (arranfw) [inline server/package-lock.json:4]: May need to take a look at your NPM/node version, this should be V1

File: client/src/pages/private/ParticipantTable.js [hunk 2]
```diff
@@ -378,117 +392,121 @@ export default () => {
...
+                variant='outlined'
+                size='small'
+                text='Archive'
+              />
+            )}
+          </>
+        );
+      case 'isIndigenous':
+        let displayValue;
+        if (row.isIndigenous === undefined) {
```
Comment (fwpushan) [inline client/src/pages/private/ParticipantTable.js:502]: This could be null. Check `null coalescing operator`

Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:502]: This should work but `isNil` might be a little safer

**Review by fwpushan — DISMISSED** [review]
LGTM, please check one minor comments.

**Review by arranfw — COMMENTED** [review]
The alignment on the CB looks a bit weird, could we center it and fix the spacing? 

image

Functionality looks good!

File: sonar-project.properties [hunk 3]
```diff
@@ -7,7 +7,7 @@ sonar.projectVersion=1.0
 
 # Path to sources
 sonar.sources=server,client/src
-sonar.exclusions=cypress
+sonar.exclusions=cypress,client/src/constants
```
Comment (fwpushan) [inline sonar-project.properties:10]: 👍

**Review by fwpushan — DISMISSED** [review]

File: client/src/pages/private/ParticipantTable.js [hunk 4]
```diff
@@ -834,24 +834,20 @@ export default () => {
...
+                onChange={() => {
+                  const newValue = reducerState?.isIndigenousFilter === 'true' ? '' : 'true';
+                  dispatch({
+                    type: 'updateKey',
+                    key: 'isIndigenousFilter',
+                    value: newValue,
+                  });
+                }}
+              />
+              <FormLabel style={{ paddingTop: '13px' }}>Indigenous participants only</FormLabel>
```
Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:850]: an id + htmlFor relation for these two would improve usability

**Review by fwpushan — APPROVED** [review]

File: client/src/pages/private/ParticipantTable.js [hunk 5]
```diff
@@ -847,7 +848,7 @@ export default () => {
                   });
                 }}
               />
-              <FormLabel style={{ paddingTop: '13px' }}>Indigenous participants only</FormLabel>
+              <FormLabel for={"isIndigenousFilterCheckbox"} style={{ paddingTop: '13px' }}>Indigenous participants only</FormLabel>
```
Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:851]: `htmlFor` is the proper React attribute

**Review by arranfw — COMMENTED** [review]
Seeing something that doesn't look right locally, looking into it now

image

**Review by arranfw — APPROVED** [review]

**Pull request merged**